From f01fdd0a55726e60f068e72573d0323129a10eb1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Luis=20Javier=20Merino=20Mor=C3=A1n?= <ninjalj@gmail.com>
Date: Thu, 10 Nov 2022 01:43:51 +0100
Subject: [PATCH] Fix trimmed selections in scrollback buffer

Commits 4e69f47b and 9046680f where using an incorrect start of buffer
when trimming trailing spaces and trailing empty cells in a line from
the scrollback buffer.  This could cause too short of a text to be
copied from the selection when there were trailing spaces in the
selection.

BUG: 461542
CCBUG: 455165
---
 src/Screen.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Screen.cpp b/src/Screen.cpp
index 2be1d612d..da220153b 100644
--- a/src/Screen.cpp
+++ b/src/Screen.cpp
@@ -2083,7 +2083,7 @@ int Screen::copyLineToStream(int line,
 
         // Exclude trailing empty cells from count and don't bother processing them further.
         // See the comment on the similar case for screen lines for an explanation.
-        while (count > 0 && (characterBuffer[start + count - 1].flags & EF_REAL) == 0) {
+        while (count > 0 && (characterBuffer[count - 1].flags & EF_REAL) == 0) {
             count--;
         }
 
@@ -2092,7 +2092,7 @@ int Screen::copyLineToStream(int line,
         } else {
             if (options.testFlag(TrimTrailingWhitespace)) {
                 // ignore trailing white space at the end of the line
-                while (count > 0 && QChar(characterBuffer[start + count - 1].character).isSpace()) {
+                while (count > 0 && QChar(characterBuffer[count - 1].character).isSpace()) {
                     count--;
                 }
             }
-- 
GitLab

